core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider.go

```diff
@@ -301,6 +311,9 @@ func (provider *GitPackageContentProvider) atomicClone(parsedURL *shared_utils.P
 		depth = depthAssumingBranchTagsCommitsAreSpecified
 	}
 
+	//TODO evaluate to use the GitHub client GetContents call instead, because we are cloning the entire repository with this approach
```

Comment:
aren't we doing a shallow clone!?

Reply:
yes, we are doing a shallow clone, there is a mistake in the comment, I was referring to the `workspace` but I put repository. I will change it.
I'm referring for instance if we want just only one file from a repository's workspace we are cloning the entire repository's workspace, but I think that we could get only that one file by using the GitHubClient.GetContents() call